tests/test_utils.py

```diff
@@ -0,0 +1,136 @@
+"""Testing utilities."""
+
+from typing import Callable, Iterable, Optional, Sequence, TypeVar, Union
+
+import numpy as np
+import numpy.typing as npt
+from decorator import decorator
+from scipy.stats.mstats import mquantiles
+from sklearn.utils import check_random_state
+
+
+def draw_bootstrap_samples(
+    values: npt.ArrayLike,
+    sample_size: npt._ShapeLike,
+    n_repeats: int,
+    random_seed: Optional[Union[int, np.random.RandomState]] = None,
+) -> Iterable[np.ndarray]:
+    """
+    Draw random samples from given values with replacement.
+
+    Parameters
+    ----------
+    values : npt.ArrayLike
+        Values to bootstrap from
+    sample_size : npt._ShapeLike
+        Size/shape of bootstrap batches
+    n_repeats : int
+        number of bootstrap batches to generate
+    random_seed : Optional[Union[int, np.random.RandomState]], optional
+        random seed, by default None
+
+    Returns
+    -------
+    Iterable[np.ndarray]
+        Iterable of bootstrap batches
+    """
+    rng = check_random_state(random_seed)
+    # rng = np.random.default_rng(rng)
+
+    samples = (
+        rng.choice(values, size=sample_size, replace=True) for _ in range(n_repeats)
+    )
+    return samples
```

Comment:
Why return a generator instead of `yield`?

Reply:
I think of them as analogous, but it seems that yield does give you nicer error messages; I'll make that change!